### ISSUE TITLE
Can not use docker_image module from OSX

### ISSUE THREAD

**Issue by BlackthornYugen** [ansible/ansible-modules-core#1450]
##### Issue Type:

Bug Report
##### Ansible Version:

```
ansible 2.0.0 (devel f55d1578d9) last updated 2015/05/25 09:58:13 (GMT -400)
  lib/ansible/modules/core: (staging 26636bbce2) last updated 2015/06/01 13:06:35 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 5187c7fcd7) last updated 2015/05/25 09:58:52 (GMT -400)
  v2/ansible/modules/core: (detached HEAD 85c8a892c8) last updated 2015/05/25 09:59:03 (GMT -400)
  v2/ansible/modules/extras: (detached HEAD 70ea058563) last updated 2015/05/25 10:06:32 (GMT -400)
  configured module search path = None
```
##### Environment:

OSX 10.10.3

```
» boot2docker version
Boot2Docker-cli version: v1.6.0
Git commit: 9894ae9
```
##### Summary:

The `docker` module itself works fine but `docker_image` tasks fail in OSX / boot2docker with the error
`ConnectionError(ProtocolError('Connection aborted.', error(2, 'No such file or directory')),)`
##### Steps To Reproduce:

The simplest task that I found to demo this was removing an image that doesn't exist.

Create the following playbook (test.yml)

```

---
- hosts: localhost
  tasks:
  - name: remove image
    docker_image: name="my/app" state=absent
```

ansible-playbook test.yml -vvvv
##### Expected Results:

This is my actual output for test.yml in Ubuntu

```
PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
<localhost> REMOTE_MODULE setup
<localhost> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1433171701.69-34147611872998 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1433171701.69-34147611872998 && echo $HOME/.ansible/tmp/ansible-tmp-1433171701.69-34147611872998']
<localhost> PUT /tmp/tmpzyiOwj TO /home/salmon/.ansible/tmp/ansible-tmp-1433171701.69-34147611872998/setup
<localhost> EXEC ['/bin/sh', '-c', u'LANG=C LC_MESSAGES=C LC_CTYPE=C /usr/bin/python /home/salmon/.ansible/tmp/ansible-tmp-1433171701.69-34147611872998/setup; rm -rf /home/salmon/.ansible/tmp/ansible-tmp-1433171701.69-34147611872998/ >/dev/null 2>&1']
ok: [localhost]

TASK: [remove image] ********************************************************** 
<localhost> REMOTE_MODULE docker_image name="my/app" state=absent
<localhost> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1433171703.39-9398745879849 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1433171703.39-9398745879849 && echo $HOME/.ansible/tmp/ansible-tmp-1433171703.39-9398745879849']
<localhost> PUT /tmp/tmpA8XzCL TO /home/salmon/.ansible/tmp/ansible-tmp-1433171703.39-9398745879849/docker_image
<localhost> EXEC ['/bin/sh', '-c', u'LANG=C LC_MESSAGES=C LC_CTYPE=C /usr/bin/python /home/salmon/.ansible/tmp/ansible-tmp-1433171703.39-9398745879849/docker_image; rm -rf /home/salmon/.ansible/tmp/ansible-tmp-1433171703.39-9398745879849/ >/dev/null 2>&1']
ok: [localhost] => {"changed": false, "failed": false, "image_id": null, "msg": ""}

PLAY RECAP ******************************************************************** 
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```
##### Actual Results:

Output from OSX

```
PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
<localhost> REMOTE_MODULE setup
<localhost> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1433171630.13-97403569354805 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1433171630.13-97403569354805 && echo $HOME/.ansible/tmp/ansible-tmp-1433171630.13-97403569354805']
<localhost> PUT /var/folders/g1/7v1x_29n01s_wd8dm8wntqxw0000gq/T/tmpscfQIO TO /Users/jsteel/.ansible/tmp/ansible-tmp-1433171630.13-97403569354805/setup
<localhost> EXEC ['/bin/sh', '-c', u'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /Users/jsteel/.ansible/tmp/ansible-tmp-1433171630.13-97403569354805/setup; rm -rf /Users/jsteel/.ansible/tmp/ansible-tmp-1433171630.13-97403569354805/ >/dev/null 2>&1']
ok: [localhost]

TASK: [remove image] ********************************************************** 
<localhost> REMOTE_MODULE docker_image name="my/app" state=absent
<localhost> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1433171630.5-193559883378401 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1433171630.5-193559883378401 && echo $HOME/.ansible/tmp/ansible-tmp-1433171630.5-193559883378401']
<localhost> PUT /var/folders/g1/7v1x_29n01s_wd8dm8wntqxw0000gq/T/tmpBV3D_I TO /Users/jsteel/.ansible/tmp/ansible-tmp-1433171630.5-193559883378401/docker_image
<localhost> EXEC ['/bin/sh', '-c', u'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /Users/jsteel/.ansible/tmp/ansible-tmp-1433171630.5-193559883378401/docker_image; rm -rf /Users/jsteel/.ansible/tmp/ansible-tmp-1433171630.5-193559883378401/ >/dev/null 2>&1']
failed: [localhost] => {"changed": false, "failed": true}
msg: ConnectionError(ProtocolError('Connection aborted.', error(2, 'No such file or directory')),)

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/Users/jsteel/test.retry

localhost                  : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by BlackthornYugen** [comment]
Another OSX user posted [a similar error on stackoverflow](https://stackoverflow.com/questions/30552317) yesterday.

**Comment by tickletik** [comment]
@BlackthornYugen  yup, that was me.

**Comment by tickletik** [comment]
I'm trying to figure out what it's looking for when it says "no such file or directory".  It's obviously breaking down over here:

`/usr/bin/python /Users/jsteel/.ansible/ ... /docker_image;`

I'm guessing the `.ansible/.../docker_image` is a python script that runs docker.  So maybe something in that file is calling up a command in a particular location, that doesn't exist at that location on OS X

**Comment by bcoca** [comment]
is your python at /usr/bin/python?

**Comment by tickletik** [comment]
@BlackthornYugen  @bcoca 

OK, I tried doing some logging in lib/python2.7/site-packages/ansible/modules/core/cloud/docker/docker.py

under class DockerManager.**init**
around line 558, they set a few environment variables:

```
    env_host = os.getenv('DOCKER_HOST')
    env_docker_verify = os.getenv('DOCKER_TLS_VERIFY')
    env_cert_path = os.getenv('DOCKER_CERT_PATH')
    env_docker_hostname = os.getenv('DOCKER_TLS_HOSTNAME')
```

So I spat out those values into a log file, and lo and behold, none of them were being set.

Then when I set them directly in main.yml:

```
- name: start container
      docker:
        name: mydata
        image: busybox
        state: present
      environment:
        DOCKER_HOST: tcp://192.168.59.103:2376
        DOCKER_TLS_VERIFY: 1
```

I got a different error:

```
failed: [localhost] => {"changed": false, "failed": true}
msg: SSLError(SSLError(SSLError(1, '_ssl.c:507: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed'),),)

FATAL: all hosts have already failed -- aborting`
```

Now docker OS X is infamous for having to use boot2docker and requiring the user to initialize the environment values.  Maybe when ansible is running it's script it is not accessing the shell values from where the ansible-playbook is being run?

**Comment by BlackthornYugen** [comment]
@bcoca Made sure python is at /usr/bin/python

```
» which python
/usr/bin/python
» file /usr/bin/python
/usr/bin/python: Mach-O universal binary with 2 architectures
/usr/bin/python (for architecture x86_64):  Mach-O 64-bit executable x86_64
/usr/bin/python (for architecture i386):    Mach-O executable i386
```

@tickletik That's interesting. I tried adding this task to see all of the environment vars: 

```
- name: check env
shell: env > env.out
```

None of my docker vars showed up in env.out. I'm not sure why it only causes problems for the docker_images module. I can start/stop/remove no problem with the docker module itself.

**Comment by tickletik** [comment]
actually I could only solve the problem with the docker module, and I had to set the environment variables directly.  So I would love to know how you managed to pass them over to that module.

docker_image however, I was totally unsuccessful in passing them over.  right now it is is setting the docker_url variable on line 223 to 'unix://var/run/docker.sock'

I don't really understand docker or ansible that well.  And i definitely don't understand TLS, etc.  So this is all confusing to me.

**Comment by tickletik** [comment]
@BlackthornYugen 

I did some more digging, and I posted my edit on stack overflow.  I'll also put my edit here.  I don't know if that is appropriate, so please delete it or ask me to delete it if you think my error description should not be here.

Following:

Digging further in docker_image.py I noticed that it breaks down around line 188 (I say "around" because I have logging breaks so I don't know the exact line) where it has the following code:

`images = self.client.images()`

so more digging and I see that self.client is checking out the docker_url at 'unix://var/run/docker.sock'

but over at [this link](https://forums.docker.com/t/docker-osx-var-run-docker-sock-file-missing/623) I see that /var/run/docker.sock does not exist on OS X, instead one reply said that

> /var/run/docker.sock will not be on your OSX filesystem - the Docker daemon is running inside the boot2docker VM - and that's where the unix socket is.
> 
> That serial file is also not related to the docker socket.
> 
> oyu need to talk to the TCP socket specified in the DOCKER_HOST env.

So I tried setting docker_url to the DOCKER_HOST URL, as the description for this modules says:

> docker_url: description: - URL of docker host to issue commands to required: false default: unix://var/run/docker.sock aliases: []

but when I set it to the DOCKER_HOST address, I got an error.

Here is main.yml

```
- name: Build docker image from dockerfiles 
      docker_image:
        name: testimage 
        path: /Users/ronny/projects/dockers/tutorial/ansibledocker/test
        state: build
        docker_url: 192.168.59.103:2376
```

and here is the error:

```
failed: [localhost] => {"changed": false, "failed": true}
msg: ConnectionError(ProtocolError('Connection aborted.', BadStatusLine('\x15\x03\x01\x00\x02\x02\n',)),)

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
       to retry, use: --limit @/Users/ronny/main.retry

localhost                  : ok=3    changed=0    unreachable=0    failed=1
```

Any ideas?

**Comment by BlackthornYugen** [comment]
When I change my docker_url I get the same result. I also tried using an invalid address (localhost), and the value of my DOCKER_HOST var (tcp://boot2docker:2376). 

It seems that in the [docker module documentation](http://docs.ansible.com/docker_module.html) the default docker_url is `${DOCKER_HOST}` OR `unix://var/run/docker.sock` but in the [docker_image module documentation](http://docs.ansible.com/docker_image_module.html) it always defaults to `unix://var/run/docker.sock`. My experiments seem to verify that the docs are accurate as to what is happening. A feature request might be to have docker_url be handled the same in both modules, but regardless of that, even when we specify the correct docker host url, we get another error. 

### Test Results:

| docker_url | ansible:devel | michaeljs1990:devel |
| :-: | --- | --- |
| _omitted_ | `msg: ConnectionError(ProtocolError('Connection aborted.', error(2, 'No such file or directory')),)` |  |
| unix://var/run/docker.sock | `msg: ConnectionError(ProtocolError('Connection aborted.', error(2, 'No such file or directory')),)` |  |
| tcp://boot2docker:2376 | `msg: ConnectionError(ProtocolError('Connection aborted.', BadStatusLine('\x15\x03\x01\x00\x02\x02\n',)),)` |  |
| tcp://192.168.59.103:2376 | `msg: ConnectionError(ProtocolError('Connection aborted.', BadStatusLine('\x15\x03\x01\x00\x02\x02\n',)),)` |  |
| tcp://localhost:2376 | `msg: ConnectionError(ProtocolError('Connection aborted.', error(61, 'Connection refused')),)` |  |

Edit: There is also a `${DOCKER_CERT_PATH}` that is referenced in the docker module docs but not docker_image docs. Maybe this is part of the issue? 
Edit 2: Made space in the table for more tests

**Comment by michaeljs1990** [comment]
I am experiencing the same error with ansible 1.9.2 and the latest version of docker-py that is in pip. Has anyone figured anything out since the last post?

**Comment by michaeljs1990** [comment]
I got this working i'll send a pull request and a write up on it tomorrow i'm about to pass out and it's pretty hacked together right now.

https://github.com/ansible/ansible-modules-core/pull/1300

Looks like this was already done. however I am doing some testing as this by itself doesn't complete fix the problem for me.

**Comment by michaeljs1990** [comment]
https://github.com/ansible/ansible-modules-core/pull/1639 was just opened and should allow you to run it if you follow the directions I have also provided. Looking for a better solution but at least we have a place to go from.

**Comment by michaeljs1990** [comment]
@BlackthornYugen If you get a chance to check out my patch that would be awesome! hope it works for you if not let me know so I can improve it.

**Comment by BlackthornYugen** [comment]
@michaeljs1990 Thanks for making that patch! I had meant to try it out yesterday but I didn't get a chance. I'll probably try it this weekend.

**Comment by michaeljs1990** [comment]
@BlackthornYugen did you get a chance to check it out over the weekend? If you could even plus 1 the PR I think it would go a long way to getting it through if they see they have a demand for it.

Thanks!

**Comment by BlackthornYugen** [comment]
@michaeljs1990 I didn't have a lot of time this weekend; but I had replaced my docker_image.py in ansible-source/v1/ansible/modules/core/cloud/docker with the one from your repo and would get an error saying that geturl isn't a valid attribute. This happens on both localhost (OSX) and my webservers (Ubuntu):

```
 _____________________ 
< TASK [remove image] >
 --------------------- 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


fatal: [localhost]: FAILED! => {  
   "invocation":{  
      "module_name":"docker_image",
      "module_args":{  
         "state":"absent",
         "name":"my/app"
      }
   },
   "failed":true,
   "traceback":"Traceback (most recent call last):\n  File \"/Users/jsteel/.ansible/tmp/ansible-tmp-1436274502.77-188547565485414/docker_image\", line 2069, in <module>\n    main()\n  File \"/Users/jsteel/.ansible/tmp/ansible-tmp-1436274502.77-188547565485414/docker_image\", line 380, in main\n    manager = DockerImageManager(module)\n  File \"/Users/jsteel/.ansible/tmp/ansible-tmp-1436274502.77-188547565485414/docker_image\", line 279, in __init__\n    base_url=docker_url.geturl(),\nAttributeError: 'str' object has no attribute 'geturl'\n",
   "parsed":false,
   "changed":false
}
fatal: [www.steelcomputers.com]: FAILED! => {  
   "msg":"Traceback (most recent call last):\r\n  File \"/home/user/.ansible/tmp/ansible-tmp-1436274502.7-241021190581477/docker_image\", line 2069, in <module>\r\n    main()\r\n  File \"/home/user/.ansible/tmp/ansible-tmp-1436274502.7-241021190581477/docker_image\", line 380, in main\r\n    manager = DockerImageManager(module)\r\n  File \"/home/user/.ansible/tmp/ansible-tmp-1436274502.7-241021190581477/docker_image\", line 279, in __init__\r\n    base_url=docker_url.geturl(),\r\nAttributeError: 'str' object has no attribute 'geturl'\r\n",
   "failed":true,
   "changed":false,
   "parsed":false,
   "invocation":{  
      "module_name":"docker_image",
      "module_args":{  
         "state":"absent",
         "name":"my/app"
      }
   }
}
fatal: [node.steelcomputers.com]: FAILED! => {
   "msg":"Traceback (most recent call last):\r\n  File \"/home/salmon/.ansible/tmp/ansible-tmp-1436274502.76-112644022138692/docker_image\", line 2069, in <module>\r\n    main()\r\n  File \"/home/salmon/.ansible/tmp/ansible-tmp-1436274502.76-112644022138692/docker_image\", line 380, in main\r\n    manager = DockerImageManager(module)\r\n  File \"/home/salmon/.ansible/tmp/ansible-tmp-1436274502.76-112644022138692/docker_image\", line 279, in __init__\r\n    base_url=docker_url.geturl(),\r\nAttributeError: 'str' object has no attribute 'geturl'\r\n",
   "failed":true,
   "changed":false,
   "parsed":false,
   "invocation":{
      "module_name":"docker_image",
      "module_args":{
         "state":"absent",
         "name":"my/app"
      }
   }
}
```

**Comment by michaeljs1990** [comment]
@BlackthornYugen huge thanks for taking a look at this! You are right I forgot to remove the .geturl method after i changed docker_image to be a string. I actually had that change locally that ended up not getting committed somehow. fixed now :) You can also add a /library folder next to your ansible playbook and place the python file in it and it will overwrite the code docker_image module to make testing easier for you.

I recently found this which has been super helpful lately http://docs.ansible.com/playbooks_best_practices.html

```
production                # inventory file for production servers
staging                   # inventory file for staging environment

group_vars/
   group1                 # here we assign variables to particular groups
   group2                 # ""
host_vars/
   hostname1              # if systems need specific variables, put them here
   hostname2              # ""

library/                  # if any custom modules, put them here (optional)
filter_plugins/           # if any custom filter plugins, put them here (optional)

site.yml                  # master playbook
webservers.yml            # playbook for webserver tier
dbservers.yml             # playbook for dbserver tier

roles/
    common/               # this hierarchy represents a "role"
        tasks/            #
            main.yml      #  <-- tasks file can include smaller files if warranted
        handlers/         #
            main.yml      #  <-- handlers file
        templates/        #  <-- files for use with the template resource
            ntp.conf.j2   #  <------- templates end in .j2
        files/            #
            bar.txt       #  <-- files for use with the copy resource
            foo.sh        #  <-- script files for use with the script resource
        vars/             #
            main.yml      #  <-- variables associated with this role
        defaults/         #
            main.yml      #  <-- default lower priority variables for this role
        meta/             #
            main.yml      #  <-- role dependencies

    webtier/              # same kind of structure as "common" was above, done for the webtier role
    monitoring/           # ""
    fooapp/               # ""
```

**Comment by BlackthornYugen** [comment]
@michaeljs1990 Thanks for the quick reply! It looks like after that update I'm back to "No such file or directory":

```
 _____________________ 
< TASK [remove image] >
 --------------------- 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


fatal: [localhost]: FAILED! => {
   "msg":"ConnectionError(ProtocolError('Connection aborted.', error(2, 'No such file or directory')),)",
   "failed":true,
   "changed":false,
   "invocation":{
      "module_name":"docker_image",
      "module_args":{
         "state":"absent",
         "name":"my/app"
      }
   }
}
ok: [node.steelcomputers.com]
ok: [www.steelcomputers.com]
```

It doesn't look like it's using DOCKER_HOST from the environment because when I specify the docker URL I get this (like I did previously): 

```
{
   "msg":"ConnectionError(ProtocolError('Connection aborted.', BadStatusLine('\\x15\\x03\\x01\\x00\\x02\\x02\\n',)),)",
   "failed":true,
   "changed":false,
   "invocation":{
      "module_name":"docker_image",
      "module_args":{
         "state":"absent",
         "docker_url":"tcp://boot2docker:2376",
         "name":"my/app"
      }
   }
}
```

**Comment by michaeljs1990** [comment]
what version of openssl are you using?

```
import ssl
print ssl.OPENSSL_VERSION
```

what is the output of `boot2docker status`?

are you able to run boot2docker ssh?

are you using python from brew or the default one with mac?

can i see the role that you are running?

Thanks! sorry for all the questions.

**Comment by michaeljs1990** [comment]
It also looks like you are doing some custom stuff such as setting boot2docker in your hosts file or at least that is what it looks like from the output.

**Comment by michaeljs1990** [comment]
this is my playbook by the way. docker still blows up with an ssl error however but docker_image is running fine.

```
---
- name: create build directory in tmp
  file: path=/tmp/docker_an_image state=directory

- name: copy id_rsa into docker build dir
  copy: src=~/.ssh/id_rsa dest=/tmp/docker_an_image/id_rsa

- name: copy docker file into directory
  copy: src=Dockerfile.animage dest=/tmp/docker_an_image/Dockerfile

- name: run docker build (This may take a few minutes)
  docker_image: tls_hostname=boot2docker path="/tmp/docker_an_image" name="an_image" state=present

- name: start image
  docker:
    name: papi
    image: an/image
    state: started
    ports:
    - "9202:80"
```

**Comment by BlackthornYugen** [comment]
@michaeljs1990

## Python / SSL Version

Don't remember if I installed a different version with brew

```
Python 2.7.6 (default, Sep  9 2014, 15:04:36) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ssl
>>> print ssl.OPENSSL_VERSION
OpenSSL 0.9.8zd 8 Jan 2015
```

## boot2docker stuff

```
» boot2docker status 
running
```

```
» boot2docker ssh
                        ##         .
                  ## ## ##        ==
               ## ## ## ## ##    ===
           /"""""""""""""""""\___/ ===
      ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
           \______ o           __/
             \    \         __/
              \____\_______/
 _                 _   ____     _            _
| |__   ___   ___ | |_|___ \ __| | ___   ___| | _____ _ __
| '_ \ / _ \ / _ \| __| __) / _` |/ _ \ / __| |/ / _ \ '__|
| |_) | (_) | (_) | |_ / __/ (_| | (_) | (__|   <  __/ |
|_.__/ \___/ \___/ \__|_____\__,_|\___/ \___|_|\_\___|_|
Boot2Docker version 1.6.0, build master : a270c71 - Thu Apr 16 19:50:36 UTC 2015
Docker version 1.6.0, build 4749651
docker@boot2docker:~$ 
```

## I did modify /etc/hosts

```
» cat /etc/hosts
##
# Host Database
#
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
##
127.0.0.1   localhost
255.255.255.255 broadcasthost
::1             localhost 
192.168.0.183   pi
192.168.59.103  docker boot2docker
```

## Playbook

Only change from the one I pasted above is that I stopped gathering facts, changed hosts to "all" and I've specified the docker_url.

```
---
- hosts: all
  gather_facts: false
  tasks:
  - name: remove image
    docker_image: name="my/app" state=absent docker_url="tcp://boot2docker:2376"
```

**Comment by michaeljs1990** [comment]
I am running 1.7.0 for boot2docker and have OpenSSL 0.9.8zf 19 Mar 2015 which I would really hope wouldn't cause an issue. I would try updating boot2docker if that is an option for you as I believe they have had quite a few bug fixes recently.

**Comment by michaeljs1990** [comment]
@BlackthornYugen do you happen to have another version of openssl installed through brew? Did you install ansible through pip or brew or pull down from source? In a second I will paste out all my environment details to see if we spot any differences.

**Comment by michaeljs1990** [comment]
@BlackthornYugen I have an odd setup but I think it's pretty standard for most mac users.

Python Setup:
I have python 2.7.10 installed through brew and python 2.7.10 which came pre installed on my mac. The brew version is installed only for the use of pip. As ansible uses #!/usr/bin/python it's running the pre installed binary. However I have $PYTHONPATH set so both versions of python are using the same pip modules. I have ansible, boto, and docker-py installed through pip. 

Boot2docker:
I have boot2docker 1.7.0 installed as specified above. I have the following set in my .bash_profile

```
export DOCKER_HOST=tcp://192.168.59.103:2376
export DOCKER_CERT_PATH=/Users/username/.boot2docker/certs/boot2docker-vm
export DOCKER_TLS_VERIFY=1
export DOCKER_OPTS="--dns 8.8.8.8 --dns 8.8.4.4"
```

since ansible runs /bin/sh when launching ssh commands that run the actual modules only .bashrc is executed so I have symlinked .bash_profile to .bashrc.

**Comment by BlackthornYugen** [comment]
@michaeljs1990 I had pulled ansible from the source. I'm not sure that I want to change my environment at the moment. But when I have a little time I'll look into upgrading to the latest boot2docker.

**Comment by jonatanblue** [comment]
I am seeing the same issue with docker-machine 0.3.0 on Mac OS X 10.10.3

**Comment by michaeljs1990** [comment]
@jonatanblue can you test out https://github.com/ansible/ansible-modules-core/pull/1639 if you have the time

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by chouseknecht** [comment]
bug_resolved

New docker modules work fine with docker-machine 1.11.2. The latest and greatest 1.12 RC release have not been tested yet. If we needed, we can open an issue for 1.12.
